src/GeniusYield/Transaction.hs

```diff
@@ -488,4 +490,34 @@ makeTransactionBodyAutoBalanceWrapper collaterals ss eh pp ps utxos body changeA
         {- Technically, this doesn't compare with the _final_ tx size, because of signers that will be
         added later. But signing witnesses are only a few bytes, so it's unlikely to be an issue -}
         Left (BuildTxSizeTooBig maxTxSize txSize)
-    first BuildTxCborSimplificationError $ simplifyGYTxBodyCbor $ txBodyFromApi txBody
+
+    collapsedBody <- first BuildTxCollapseExtraOutError $ collapseExtraOut extraOut txBodyContent
+
+    first BuildTxCborSimplificationError $ simplifyGYTxBodyCbor $ txBodyFromApi collapsedBody
+
+collapseExtraOut
+  :: forall era
+  . Api.S.IsCardanoEra era
+  => Api.TxOut Api.S.CtxTx era
+  -> Api.TxBodyContent Api.S.BuildTx era
+  -> Either Api.S.TxBodyError (Api.S.TxBody era)
+collapseExtraOut apiOut@(Api.TxOut outAddr outVal _ _) (bodyContent@(Api.TxBodyContent {txOuts})) =
+    case delete apiOut cOuts of
```

Comment:
Why `delete` here? Isn't the change output generated (if any) simply the last one in outputs?
![image](https://github.com/geniusyield/atlas/assets/29835247/371254c3-3175-4022-9b48-79d2ae61b7a0)

Image source: [here](https://input-output-hk.github.io/cardano-node/cardano-api/lib/src/Cardano.Api.Fees.html#local-6989586621679608579).

Reply:
So, just need to check if the last output is same as this given change output.

Reply:
I agree that we should only merge into outputs generated by coin selection, and not all outputs that have the same change address. 

Regarding the creation or not of the extra output, it will always be created, unless the balance between inputs and outputs + fees is exactly 0. (If it's negative or below the minAda requirement, it will throw an error) In that case, the output will not be added to the transaction, but we will still receive a TxOut (with a 0 Lovelace value). So in that case, we can still select an output and add the values together without affecting the final tx. Of course we could also check if the balance is 0 as a first step and return immediately if that's the case.